### DISCUSSION TITLE
exposing more methods in the `types.Transaction` object ?

### DISCUSSION THREAD

**Discussion by buixor** [corazawaf/coraza#784] [Q&A]
Hello,

First of all, thanks for the cool project. I have a few questions!

While playing around with coraza (v3), I found the need to be able to use, for example, `RemoveRuleByID` at runtime.

However, it seems that the `types.Transaction` object that you get from the `NewTransaction` method doesn't offer much "runtime configuration" methods, that **do** exist in the underlying internal `Transaction` object. For now, I have forked coraza to expose the `RemoveRuleByID` method in `types.Transaction` and it didn't seem to implode.

Q: Am I missing something, is it intentional, or is it already planned?

**Comment by jptosso** [comment]
Hey! Seclang is designed to allow runtime modifications using ctl, that’s why we don’t see reason to provide mechanisms to alter rule flows programmatically. It is open to discussion if there is a cool use case though

**Reply by buixor** [reply]
Hello,

If I am correct, ctl can only be used at configuration time?

We are looking to integrate Coraza within crowdsec, and we want to allow the user to use the expr integration to finetune the engine at runtime. The immediate thing we are thinking of is, of course, disabling some rules at runtime based on the context (ie. path is .. / IP belongs to range ... etc.).

I am happy to provide PR if this can help this topic.
